api/android/api/src/main/java/org/nnsuite/nnstreamer/SingleShot.java

```diff
@@ -83,30 +84,55 @@ public SingleShot(@NonNull File model, @Nullable TensorsInfo in, @Nullable Tenso
 
     /**
      * Invokes the model with the given input data.
-     * Even if the model has flexible input data dimensions,
-     * input data frames of an instance of a model should share the same dimension.
+     * If the model has flexible input data dimensions, input information for this
+     * run of the model can be passed. This changes the currently set input information
+     * for this instance of the model. The corresponding output information can be
+     * extracted.
      *
      * Note that this has a default timeout of 3 seconds.
      * If an application wants to change the time to wait for an output,
      * set the timeout using {@link #setTimeout(int)}.
      *
      * @param in The input data to be inferred (a single frame, tensor/tensors)
+     * @param inInfo The input tensors information
+     * @param outInfo The container for output tensors information
      *
      * @return The output data (a single frame, tensor/tensors)
      *
      * @throws IllegalStateException if failed to invoke the model
      * @throws IllegalArgumentException if given param is null
      */
-    public TensorsData invoke(@NonNull TensorsData in) {
+    public TensorsData invoke(@NonNull TensorsData in, @Nullable TensorsInfo inInfo, @Nullable TensorsInfo outInfo) {
```

Comment:
plz add another method with the params (TensorsData in, TensorsInfo inInfo).
you may not update outInfo in this method.

Reply:
> plz add another method with the params (TensorsData in, TensorsInfo inInfo).
> you may not update outInfo in this method.

Did you mean another function which will return `TensorsInfo`?
If yes, then both the functions will have same set of parameters, and in java, two functions with same name+paramters and different output type is not allowed.
How about passing `outInfo` as a pointer, and set `outInfo` to output from `getOutputInfo()`?

Reply:
I mean two invoke  methods:

```
public TensorsData invoke(@NonNull TensorsData in) {
    return invoke(in, null);
}

public TensorsData invoke(@NonNull TensorsData in, @Nullable TensorsInfo inInfo) {
  ...
  // we cannot update output-info (passed arg is variable, not pointer)
}

```

Reply:
In these two definitions, `inInfo` is `nullable`, the second function is automatically a superset of the first function. If we want to have both definitions, then `inInfo` should be `NonNull`. 
If we want `Nullable`, then the only the second method can cover both of them.

Reply:
Developer can set (or update) dimension with following steps:

1. open with changed dimension 
SingleShot single = new SingleShot (model, in, out); <-- input and output dimension can be changed during this method call

single.invoke (data);

2. open with default dimension
SingleShot single = new SingleShot(model);

single.setInputInfo (inInfo); <-- input and output dim will be changed

single.invoke (data);

Reply:
I added `setInputInfo`, and an updated invoke which is a mixed version of `invoke`, and `dynamicInvoke` and can be used as per needed.